docs/source/json-api/index.rst

```diff
@@ -21,12 +21,15 @@ complicating concerns including, but not limited to:
 - asynchronous submit/completion workflows,
 - temporal queries (e.g. active contracts *as of a certain time*), and
 
-For these and other features, use :doc:`the Ledger API </app-dev/ledger-api>`
-instead.
+For these and other features, use :doc:`the Ledger API </app-dev/ledger-api>` instead.
+The JSON API is a "proxy", after a fashion, for that API; *there is literally nothing that JSON API can do that your own application cannot do via gRPC*.
 
 If you are using this API from JavaScript or TypeScript, we strongly recommend using :doc:`the JavaScript bindings and code generator </app-dev/bindings-ts/index>` rather than invoking these endpoints directly.
```

Comment:
Should we not also recommend to use the Java API and code generator instead of this JSON API?

Reply:
This is the suggestion *if you are using this API from JavaScript or TypeScript*; the Java bindings/codegen are irrelevant in the case of this premise.

I separately added a suggestion of that kind.